Week 10/RecipeApp/app/src/main/java/com/janewaitara/movieapp/worker/SynchronizeDataWorker.kt

```diff
@@ -0,0 +1,41 @@
+package com.janewaitara.movieapp.worker
+
+import android.content.Context
+import android.widget.Toast
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.janewaitara.movieapp.RecipeApplication
+import com.janewaitara.movieapp.model.Success
+import com.janewaitara.movieapp.repository.RoomRepository
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+
+/**
+ *Synchronize data every e.g. 1 hour. Use a PeriodicWorkRequest to implement this type of behavior.
+ * This worker load data from the API, and then store it in the Database.
+ */
+
+/**
+ *Implement KoinComponent interface as a way to tell koin that it can inject fields into this class
+ * (which is not a lifeCycle Class)*/
+class SynchronizeDataWorker(context: Context, workerParameters: WorkerParameters):
+    CoroutineWorker(context,workerParameters), KoinComponent {
+
+    private val repository: RoomRepository by inject()
```

Comment:
Nice job using the injection here as well. Filip and Luka mentioned that generally the dependency is injected as the final dependency, for example the viewModel would be injected into the fragment or activity, and the rest of the injection throughout the project would be acheived via constructor injection. However, the video course showed a slightly different pattern so this is not included in the scoring at all, I just wanted to mention it for future knowledge.